src/main/java/org/graylog/integrations/aws/resources/requests/AWSInputCreateRequest.java

```diff
@@ -76,20 +60,38 @@
     @JsonProperty(ADD_FLOW_LOG_PREFIX)
     public abstract boolean addFlowLogPrefix();
 
-    @JsonCreator
-    public static AWSInputCreateRequest create(@JsonProperty(NAME) String name,
-                                               @JsonProperty(AWS_MESSAGE_TYPE) String awsMessageType,
-                                               @JsonProperty(AWSRequest.AWS_ACCESS_KEY_ID) String awsAccessKey,
-                                               @JsonProperty(AWSRequest.AWS_SECRET_ACCESS_KEY) String awsSecretKey,
-                                               @JsonProperty(STREAM_NAME) String streamName,
-                                               @JsonProperty(AWSRequest.REGION) String region,
-                                               @JsonProperty(BATCH_SIZE) int batchSize,
-                                               @JsonProperty(ASSUME_ROLE_ARN) String assumeRoleArn,
-                                               @JsonProperty(GLOBAL) boolean global,
-                                               @JsonProperty(THROTTLING_ALLOWED) boolean enableThrottling,
-                                               @JsonProperty(ADD_FLOW_LOG_PREFIX) boolean addFlowLogPrefix) {
-        return new AutoValue_AWSInputCreateRequest(name, awsMessageType, awsAccessKey, awsSecretKey,
-                                                   streamName, assumeRoleArn, region, batchSize, global,
-                                                   enableThrottling, addFlowLogPrefix);
+    public static Builder builder() {
+        return Builder.create();
+    }
+
+    @AutoValue.Builder
+    public static abstract class Builder implements AWSRequest.Builder<Builder> {
```

Comment:
@danotorrey Where would a Builder like this come in handy ? Can I re-purpose this builder else where within ipfixcodec.java to create objects of InformationElementDefinitions .java ?